etl/grapher_model.py

```diff
@@ -1,23 +1,371 @@
+```
+It has been slightly modified since then.
+"""
 import json
-from typing import Any, Dict, Optional
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+from urllib.parse import quote
 
 import pandas as pd
 import structlog
 from pydantic import BaseModel
-from sqlalchemy.engine import Engine
-from sqlmodel import JSON, Column, Field, Session, SQLModel, select
+from sqlalchemy import (
+    TIMESTAMP,
+    BigInteger,
+    Computed,
+    DateTime,
+    ForeignKeyConstraint,
+    Index,
+    Integer,
+    String,
+    Table,
+    text,
+)
+from sqlalchemy.dialects.mysql import (
+    INTEGER,
+    LONGBLOB,
+    LONGTEXT,
+    MEDIUMTEXT,
+    TINYINT,
+    VARCHAR,
+)
+from sqlalchemy.future import Engine as _FutureEngine
+from sqlmodel import (
+    JSON,
+    Column,
+    Field,
+    Relationship,
+    Session,
+    SQLModel,
+    create_engine,
+    select,
+)
 from sqlmodel.sql.expression import Select, SelectOfScalar
 
-log = structlog.get_logger()
+from etl import config
 
+log = structlog.get_logger()
 
 # get rid of warning from https://github.com/tiangolo/sqlmodel/issues/189
 SelectOfScalar.inherit_cache = True  # type: ignore
 Select.inherit_cache = True  # type: ignore
 
+metadata = SQLModel.metadata
+
+
+def get_engine() -> _FutureEngine:
+    return create_engine(
+        f"mysql://{config.DB_USER}:{quote(config.DB_PASS)}@{config.DB_HOST}:{config.DB_PORT}/{config.DB_NAME}"
+    )
+
+
+t_active_datasets = Table(
+    "active_datasets",
+    metadata,
+    Column("id", Integer, server_default=text("'0'")),
+    Column("name", String(512)),
+    Column("description", LONGTEXT),
+    Column("createdAt", DateTime),
+    Column("updatedAt", DateTime),
+    Column("namespace", String(255)),
+    Column("isPrivate", TINYINT(1), server_default=text("'0'")),
+    Column("createdByUserId", Integer),
+    Column("metadataEditedAt", DateTime),
+    Column("metadataEditedByUserId", Integer),
+    Column("dataEditedAt", DateTime),
+    Column("dataEditedByUserId", Integer),
+    Column("nonRedistributable", TINYINT(1), server_default=text("'0'")),
+    Column("isArchived", TINYINT(1), server_default=text("'0'")),
+    Column("sourceChecksum", String(64)),
+)
+
+
+class CountryNameToolContinent(SQLModel, table=True):
+    __table_args__ = (
+        Index("continent_code", "continent_code", unique=True),
+        Index("continent_name", "continent_name", unique=True),
+    )
+
+    id: Optional[int] = Field(default=None, sa_column=Column("id", Integer, primary_key=True))
+    continent_code: str = Field(sa_column=Column("continent_code", String(255, "utf8mb4_0900_as_cs"), nullable=False))
+    continent_name: str = Field(sa_column=Column("continent_name", String(255, "utf8mb4_0900_as_cs"), nullable=False))
+
+    country_name_tool_countrydata: List["CountryNameToolCountrydata"] = Relationship(
+        back_populates="country_name_tool_continent"
+    )
+
+
+class Details(SQLModel, table=True):
```

Comment:
Even though they're autogenerated, I had to make some changes in the next PR. I've decided to leave them here because other models that we actively use can reference them in `Relationship` (and I wanted to have the complete specification of tables we use).

> so it would seem the fewer models we include the better.

I'll go with a compromise and remove models with no chance of being used ever (like `KnexMigrations`) and keep those that could be useful (like `ChartRevisions`). I don't have strong feelings about this, but if this annoys you at some point, feel free to remove unused models.